backend/Squirrel.Core/Squirrel.Core.WebAPI/Extentions/ServiceCollectionExtensions.cs

```diff
@@ -1,23 +1,40 @@
-﻿using Squirrel.Core.BLL.MappingProfiles;
+﻿using AutoMapper;
+using FluentValidation.AspNetCore;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.BLL.MappingProfiles;
 using Squirrel.Core.BLL.Services;
+using Squirrel.Core.Common.Models;
 using Squirrel.Core.DAL.Context;
-using Squirrel.Core.BLL.Interfaces;
+using Squirrel.Core.DAL.Entities;
 using Squirrel.Core.WebAPI.Validators;
-using FluentValidation.AspNetCore;
-using Microsoft.EntityFrameworkCore;
 using System.Reflection;
 
 namespace Squirrel.Core.WebAPI.Extentions
 {
     public static class ServiceCollectionExtensions
     {
-        public static void RegisterCustomServices(this IServiceCollection services)
+        public static void RegisterCustomServices(this IServiceCollection services, IConfiguration configuration)
         {
             services
                 .AddControllers()
                 .AddNewtonsoftJson(options => options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore);
 
             services.AddTransient<ISampleService, SampleService>();
+
+            services.Configure<MongoDatabaseConnectionSettings>(
+                configuration.GetSection("MongoDatabase"));
+
+            services.AddTransient(typeof(IMongoService<Sample>), serviceProvider =>
+            {
+                var context = serviceProvider.GetRequiredService<SquirrelCoreContext>();
+                var mapper = serviceProvider.GetRequiredService<IMapper>();
+                var mongoDatabaseSettings = serviceProvider.GetRequiredService<IOptions<MongoDatabaseConnectionSettings>>();
+                string collectionName = "SampleCollection";
+
+                return new MongoService<Sample>(mongoDatabaseSettings, collectionName);
+            });
```

Comment:
It would probably be better to move this part to a separate extension method `AddMongoDbService`. Also why are you getting context and mapper here?

Reply:
Oh, my bad. I forgot to remove it. Thanks!